internal/manifests/gateway/gateway.go

```diff
@@ -58,6 +58,25 @@ func BuildGateway(params manifestutils.Params) ([]client.Object, error) {
 
 	dep := deployment(params, rbacCfgHash, tenantsCfgHash)
 
+	if params.Gates.HTTPEncryption || params.Gates.GRPCEncryption {
+		caBundleName := naming.SigningCABundleName(params.Tempo.Name)
+		if err := manifestutils.ConfigureServiceCA(&dep.Spec.Template.Spec, caBundleName); err != nil {
+			return nil, err
+		}
+		err := manifestutils.ConfigureServicePKI(params.Tempo.Name, manifestutils.DistributorComponentName, &dep.Spec.Template.Spec)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if params.Gates.HTTPEncryption {
+		dep.Spec.Template.Spec.Containers[0].Args = append(dep.Spec.Template.Spec.Containers[0].Args,
+			fmt.Sprintf("--traces.tls.key-file=%s/tls.key", manifestutils.TempoServerTLSDir()),
```

Comment:
I like how for the query-frontend you modified the deployment in the `deployment()` function, I'd suggest doing the same here also, i.e. moving the code which modifies the deployment spec to the `deployment()` function.

In case someone is modifying the deployment spec, the chances are higher to also modify this code (if needed) if they're in the same function and not in different functions.